src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3144,10 +3144,6 @@ public IndexDefinition GetIndexDefinition()
         {
             QueryInternalPreparation(query);
 
-            if (resultToFill.SupportsInclude == false
-                && (query.Metadata.Includes != null && query.Metadata.Includes.Length > 0))
-                throw new NotSupportedException("Includes are not supported by this type of query.");
```

Comment:
What about other checks that we have here? For example highlighting is not supported by collection query as well. Will we throw? Another question -`SupportsInclude` is it used anywhere else?

Please ping me about that so we'll talk directly.

Reply:
Moved all checks to one function so we can reuse it